### PR TITLE
User/shasnayak/port object lifetime tests

### PR DESCRIPTION
**Pull request by ShashankNay**
This change introduces a series of Object Lifetime tests that detect memory leaks using MSTest.

### CODE REVIEW

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/UnitTest.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace ObjectLifetimeTests.Lifted
+{
+    [TestClass]
+    public class UnitTest
+    {
+        [TestMethod]
+        public void TestMethod1()
+        {
+            Assert.AreEqual(0, 0);
```
Comment (azchohfi) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/UnitTest.cs:16]: I don't think this file is being used, or needed.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            <ExcludeAssets>build</ExcludeAssets>
+        </PackageReference>
+        <PackageReference Include="MSTest.TestAdapter">
+            <Version>2.2.4-preview-20210513-02</Version>
+        </PackageReference>
+        <PackageReference Include="MSTest.TestFramework">
+            <Version>2.2.4-preview-20210513-02</Version>
+        </PackageReference>
+        <!-- We have to bump the Newtonsoft.Json version temporarily, until version is bumped on Microsoft.TestPlatform.TestHost -->
+        <PackageReference Include="Newtonsoft.Json">
```
Comment (azchohfi) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj:78]: This dependency is not needed anymore.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        </PackageReference>
+        <PackageReference Include="Microsoft.TestPlatform.TestHost">
+            <Version>17.0.0-dev</Version>
+            <ExcludeAssets>build</ExcludeAssets>
+        </PackageReference>
+        <PackageReference Include="Microsoft.ProjectReunion" Version="0.5.0-20210324.14.CI" />
+        <PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.5.0-20210324.14.CI" />
+        <PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="3.0.0-20210324.14.CI" />
+        <PackageReference Include="coverlet.collector" Version="1.3.0" />
+        <PackageReference Include="Newtonsoft.Json">
```
Comment (azchohfi) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj:27]: Not needed anymore.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted (Package)/ObjectLifetimeTests.Lifted (Package).wapproj [hunk 4]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (codendone) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted (Package)/ObjectLifetimeTests.Lifted (Package).wapproj:2]: Are any of the files in the "(Package)" directory needed? I thought these tests worked without using the WAP?

File: src/Tests/ObjectLifetimeTests/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<Project>
```
Comment (codendone) [inline src/Tests/ObjectLifetimeTests/Directory.Build.props:1]: Are these empty files needed? Is this overriding default files we don't want?

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted (Package)/ObjectLifetimeTests.Lifted (Package).wapproj (hunk 4 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted (Package)/ObjectLifetimeTests.Lifted (Package).wapproj]: They do, I wasn't sure whether or not the WAP was still necessary to get them running. Tested this locally and the tests succeed without the wapproj-related items so I've removed them.

File: src/Tests/ObjectLifetimeTests/Directory.Build.props (hunk 5 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Directory.Build.props:1]: Yes, it's to prevent inheriting items we don't want.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj [hunk 6]
```diff
@@ -0,0 +1,118 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <TargetFramework>net5.0-windows10.0.19041.0</TargetFramework>
+        <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
+        <RootNamespace>ObjectLifetimeTests.Lifted</RootNamespace>
+        <ApplicationManifest>app.manifest</ApplicationManifest>
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj:7]: Can we make these tests use the Windows SDK and WinUI projection we build in the cswinrt repo (the ones under the projection folder)

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/nuget.config [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/nuget.config:3]: Do we need this nuget.config or is the default fine?

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+      <Link>Tests\RealMainPage.xaml</Link>
+      <Generator>MSBuild:Compile</Generator>
+      <SubType>Designer</SubType>
+    </Page>
+    <Page Include="MainPage.xaml">
+      <Generator>MSBuild:Compile</Generator>
+      <SubType>Designer</SubType>
+    </Page>
+  </ItemGroup>
+  <ItemGroup>
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj:208]: Do adding WUX lifetime tests make sense here?

File: src/Tests/ObjectLifetimeTests/Tests/CycleTestCanvas.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+#endif
+
+namespace ObjectLifetimeTests
+{
+    public class CycleTestCanvas : Canvas
+    {
+        public CycleTestCanvas()
+        {
+        }
+//#if !VSPROJ
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/CycleTestCanvas.cs:24]: is ifdef needed?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    d:DesignHeight="300"
+    d:DesignWidth="400">
+
+    <Canvas x:Name="_root" >
+
+        <!-- Create a tree of non-visual objects -->
+        <!--<Canvas.Background>
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml:15]: Is this used?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs [hunk 11]
```diff
@@ -0,0 +1,1487 @@
+﻿
+//
+// This is copied from ObjectLifetimeTests.cs in the OS repo for RS5
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:3]: Is this comment needed?

File: src/Tests/UnitTest/UnitTest.csproj [hunk 12]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <!--Target .NET Core 2.0 to test .NET Standard 2.0 projection -->
-    <TargetFrameworks>netcoreapp2.0;net5.0</TargetFrameworks>
+    <TargetFrameworks>net5.0</TargetFrameworks>
```
Comment (manodasanW) [inline src/Tests/UnitTest/UnitTest.csproj:5]: Is this change intentional?   I assume not.

File: src/build.cmd [hunk 13]
```diff
@@ -1,8 +1,8 @@
 @echo off
 if /i "%cswinrt_echo%" == "on" @echo on
 
-set CsWinRTNet5SdkVersion=5.0.100
-
+set CsWinRTNet5SdkVersion=5.0.104
+set msbuild_verbosity=diag
```
Comment (manodasanW) [inline src/build.cmd:5]: probably don't want msbuild_verbosity here.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.sln [hunk 14]
```diff
@@ -0,0 +1,81 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.31025.194
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{C7167F0D-BC9F-4E6E-AFE1-012C56B48DB5}") = "ObjectLifetimeTests.Lifted (Package)", "ObjectLifetimeTests.Lifted (Package)\ObjectLifetimeTests.Lifted (Package).wapproj", "{AA56B739-3D4C-47DD-B4F0-5139DE01F291}"
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.sln:6]: IF the csproj is the cswinrt.sln, is this separate sln still needed?

File: src/Tests/ObjectLifetimeTests/Directory.Build.props (hunk 5 above)
Comment (codendone) [reply src/Tests/ObjectLifetimeTests/Directory.Build.props:1]: Worh adding a comment to clarify that?

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj (hunk 6 above)
Comment (Scottj1s) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj:7]: Shashank and I chatted about how to do this.  I suggested renaming (via AssemblyName project prop) our test projection to Microsoft.WinUI.dll, possibly setting major version to something artificially high (e.g., 99), and possibly strong name signing.  That should match link requirements of MSTest and ensure that the later version overrides.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/Package.appxmanifest [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="3fb45547-1ef1-4576-a2b9-3be5d5659cca"
+    Publisher="CN=mikeh"
+    Version="1.0.0.0" />
+
+  <Properties>
+    <DisplayName>ObjectLifetimeTests.Lifted (Package)</DisplayName>
+    <PublisherDisplayName>mikeh</PublisherDisplayName>
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/Package.appxmanifest:16]: remove mikeh

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/Package.appxmanifest [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="3fb45547-1ef1-4576-a2b9-3be5d5659cca"
+    Publisher="CN=mikeh"
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/Package.appxmanifest:11]: remove mikeh (just make something up like 'cswinrt')

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/nuget.config (hunk 7 above)
Comment (Scottj1s) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/nuget.config:3]: yes, this should not be needed

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/App.xaml.cs [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+    sealed partial class App : Application
+    {
+        /// <summary>
+        /// Initializes the singleton application object.  This is the first line of authored code
+        /// executed, and as such is the logical equivalent of main() or WinMain().
+        /// </summary>
+        public App()
+        {
+            this.InitializeComponent();
+            this.Suspending += OnSuspending;
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/App.xaml.cs:32]: the suspending event is experimental - should remove it (and handler)

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj (hunk 8 above)
Comment (Scottj1s) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj:208]: we should open a separate issue for additional tests, as there are dependencies on this PR

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  IgnorableNamespaces="uap mp">
+
+  <Identity
+    Name="2cbb106a-c4fb-4dc8-8e72-f0727f24f544"
+    Publisher="CN=mikeh"
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest:11]: mikeh

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+    Version="1.0.0.0" />
+
+  <mp:PhoneIdentity PhoneProductId="2cbb106a-c4fb-4dc8-8e72-f0727f24f544" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
+
+  <Properties>
+    <DisplayName>ObjectLifetimeTests0325</DisplayName>
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest:17]: what is the 0325 suffix?

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+    <PublisherDisplayName>mikeh</PublisherDisplayName>
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/Package.appxmanifest:18]: mikeh

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj (hunk 8 above)
Comment (manodasanW) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Wux/ObjectLifetimeTests.Wux.csproj:208]: This comment wasn't regarding adding additional tests, but rather this file is for tests that are testing WUX, not MUX.

File: src/Projections/Reunion/Reunion.csproj [hunk 21]
```diff
@@ -0,0 +1,53 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Scottj1s) [inline src/Projections/Reunion/Reunion.csproj:1]: we shouldn't add a new projection - just update the existing WinUI projection to use Reunion

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+        <Platforms>x86;x64;arm64</Platforms>
+        <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
+        <EnablePreviewMsixTooling>True</EnablePreviewMsixTooling>
+        <PublishProfile>win10-$(Platform).pubxml</PublishProfile>
+        <PrepareMsixPackageDependsOn></PrepareMsixPackageDependsOn>
+        <PublishReadyToRun>false</PublishReadyToRun>
+        <ProjectReunionVerifyWinrtRuntimeVersion>false</ProjectReunionVerifyWinrtRuntimeVersion>
+    </PropertyGroup>
+    <PropertyGroup>
+        <DefineConstants>LIFTED</DefineConstants>
```
Comment (Scottj1s) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted/ObjectLifetimeTests.Lifted.csproj:17]: I doubt we'll ever test system in C#/WinRT - can we remove the LIFTED define and all conditional code to simplify?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml (hunk 10 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml:15]: Doesn't look like it. Removed

File: src/build.cmd [hunk 23]
```diff
@@ -121,6 +121,8 @@ if not exist %nuget_dir%\nuget.exe powershell -Command "Invoke-WebRequest https:
 rem Note: packages.config-based (vcxproj) projects do not support msbuild /t:restore
 call %this_dir%get_testwinrt.cmd
 call :exec %nuget_dir%\nuget.exe restore %nuget_params% %this_dir%cswinrt.sln
+rem: Calling nuget restore again on ObjectLifetimeTests.Lifted.csproj to prevent .props from \microsoft.testplatform.testhost\build\netcoreapp2.1 from being included. Nuget.exe erroneously imports props files.
```
Comment (RealTommyKlein) [inline src/build.cmd:124]: Would also add a link to the bug causing this, NuGet/Home#9672

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+            <Version>2.2.4-preview-20210412-01</Version>
+        </PackageReference>
+        <PackageReference Include="MSTest.TestFramework">
+            <Version>2.2.4-preview-20210412-01</Version>
+        </PackageReference>
+        <PackageReference Include="Microsoft.TestPlatform.TestHost">
+            <Version>17.0.0-dev</Version>
+            <ExcludeAssets>build</ExcludeAssets>
+        </PackageReference>
+        <PackageReference Include="Microsoft.ProjectReunion" Version="0.5.0-20210324.14.CI" />
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj:23]: Are these release builds?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+
+    <Canvas x:Name="_root" >
+    
+        <Canvas x:Name="MoveCanvas" Canvas.Top="250" >
+            <Rectangle Width="200" Height="50" Fill="Blue" />
+        </Canvas>
+
+        <ItemsControl Name='_itemsControlLeakTest'>
+            <Rectangle Width="100" Height="100" Fill="Red" />
+            <!--<local:MyControl  />-->
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml:20]: intentional?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs (hunk 11 above)
Comment (manodasanW) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:3]: Thoughts on addressing this?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs [hunk 26]
```diff
@@ -0,0 +1,1483 @@
...
+        WeakReference _element1Ref = null;
+        WeakReference _element2Ref = null;
+        object _element1 = null;
+        object _element2 = null;
+
+
+        AutoResetEvent loadedSignal = new AutoResetEvent(false);
+        AutoResetEvent unloadedSignal = new AutoResetEvent(false);
+
+        //[TestMethod]
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:368]: Do [Ignore] work?  Might be better to use as it allows the test to still compile.

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs [hunk 27]
```diff
@@ -0,0 +1,1483 @@
...
+            _asyncQueue.Run();
+        }
+
+        //
+        // Cycle:
+        // CustomDependencyObject
+        // => Binding(CustomDependencyObject.AttachedObject = Button.Tag)
+        // => Button.Tag = CustomDependencyObject
+        //
+        //[Disabled("595913", "CodeDefect")] // resolved Won't Fix
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:1057]: Remove?  I assume this works now?

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs (hunk 26 above)
Comment (manodasanW) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:368]: or by commenting out TestMethod attribute as we do below in a couple other tests.

File: src/VsTestResults.trx [hunk 28]
```diff
@@ -0,0 +1,812 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TestRun id="140d787a-23c7-478d-b9c8-eeb344604aa8" name="shasnayak@DESKTOP-V5VHPHF 2021-06-15 18:56:36" runUser="NORTHAMERICA\shasnayak" xmlns="http://microsoft.com/schemas/VisualStudio/TeamTest/2010">
+  <Times creation="2021-06-15T18:56:36.4935878-04:00" queuing="2021-06-15T18:56:36.4935878-04:00" start="2021-06-15T18:56:23.2865498-04:00" finish="2021-06-15T18:56:36.6940195-04:00" />
```
Comment (manodasanW) [inline src/VsTestResults.trx:3]: Is this the run result file?  Should we not check this in?

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj [hunk 29]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj:2]: Is this file used?  If not, do you think it would be easy to just merge the other 2 folders (test and ObjectLifetimeTests.Lifted)

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/Directory.Build.props [hunk 30]
```diff
@@ -0,0 +1,8 @@
+<Project>
+    <Import Project="ObjectLifetimeTests.Lifted\CSWinRTInApp.props"   />
+    <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+    <Import Project="..\..\..\..\nuget\Microsoft.Windows.CsWinRT.props" />
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/Directory.Build.props:4]: Any reason why the one in the root directory props doesn't automatically apply here?  Is it because we don't import directory.build.props at a certain folder level and if so why?

Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Lifted/Directory.Build.props:4]: The reason was because the `SimulateCsWinRTNugetReference` property was set after the import statement. After adding the property group before the import statement there no longer needs to be a direct reference to the `Microsoft.Windows.CsWinRT.props` file.

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs (hunk 26 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:368]: I believe commenting out the TestMethod attribute should be enough to prevent it from running.

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs (hunk 27 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs:1057]: Still a failing test, the absence of `[TestMethod]`  made it not run.

File: src/Tests/ObjectLifetimeTests/Directory.Build.targets [hunk 31]
```diff
@@ -1,2 +1,5 @@
 <Project>
+    <Import Project="CSWinRTInApp.targets"/>
+    <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(MSBuildThisFileDirectory)../'))" />    
+    <Import Condition="'$(MSBuildProjectExtension)' == '.csproj' and '$(SimulateCsWinRTNugetReference)' == 'true'" Project="..\..\..\nuget\Microsoft.Windows.CsWinRT.targets" />
```
Comment (manodasanW) [inline src/Tests/ObjectLifetimeTests/Directory.Build.targets:4]: Would the same apply here that this comes from the import?

**Review by manodasanW — APPROVED** [review]

File: src/Tests/ObjectLifetimeTests/Directory.Build.targets (hunk 31 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Directory.Build.targets]: Yes, removed.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj (hunk 29 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj]: Unused, removed and merged folders.

File: src/VsTestResults.trx (hunk 28 above)
Comment (ShashankNay) [reply src/VsTestResults.trx]: Deleted this

File: src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml (hunk 25 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimePage.xaml]: Unintentional, removed.

File: src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj (hunk 24 above)
Comment (ShashankNay) [reply src/Tests/ObjectLifetimeTests/ObjectLifetimeTests.Tests/ObjectLifetimeTests.Tests.csproj]: Removed
